app/models/miq_server/worker_management/monitor.rb

```diff
@@ -26,7 +26,18 @@ def monitor_workers
     processed_worker_ids += check_not_responding
     processed_worker_ids += check_pending_stop
     processed_worker_ids += clean_worker_records
-    processed_worker_ids += request_workers_to_sync_config(resync_needed)
+
+    # Monitor all remaining current worker records
+    processed_worker_ids += miq_workers.where(:status => MiqWorker::STATUSES_CURRENT_OR_STARTING).each do |worker|
```

Comment:
Are the processed_worker_ids actually ids or records?  by adding in the where clause aren't you mixing 2 different types?

Reply:
Gah, I totally read this as `(_ids + query).each { } .map(&:ids)` as opposed to `_ids + query.map(&:ids)`.  Ignore

Reply:
I map to ids after the `each` block's `end`

This is all going away after I rebase #19585 onto this so it's gross, but very temporary.

Reply:
Ah, also they're ids https://github.com/ManageIQ/manageiq/pull/19609/files#diff-70e718c4e35ca2fea4d16f7312ab7a93R84 FWIW